src/System.Device.Gpio/System/Device/Gpio/GpioController.cs

```diff
@@ -115,6 +118,20 @@ public void OpenPin(int pinNumber, PinMode mode)
             SetPinMode(pinNumber, mode);
         }
 
+        /// <summary>
+        /// Opens a pin and sets it to a specific mode and value.
+        /// </summary>
+        /// <param name="pinNumber">The pin number in the controller's numbering scheme.</param>
+        /// <param name="mode">The mode to be set.</param>
+        /// <param name="initialValue">The initial value to be set if the mode is output. The driver will attempt to set the mode without causing glitches to the other value.
+        /// (if <paramref name="initialValue"/> is <see cref="PinValue.High"/>, the pin should not glitch to low during open)</param>
+        public void OpenPin(int pinNumber, PinMode mode, PinValue initialValue)
+        {
+            OpenPin(pinNumber);
```

Comment:
Should we instead have an OpenPin in the driver that takes in an initial value? That way if there is a driver that supports setting an initial value during opening they can do it right there, while other drivers that don't can make the two operations separately like you are doing here.

Reply:
I think that's not required, because `OpenPin(pinNumber)` should not change the pin mode or the value on open. This would be different if we called `OpenPin(pinNumber, mode)` here.